src/xml/gff3creator.cpp

```diff
@@ -0,0 +1,194 @@
+			if (strctNode->findChild("text") == 0)
+				strctPtr->addExoString(strctNode->getProperty("label"), "");
+			else
+				strctPtr->addExoString(strctNode->getProperty("label"), strctNode->findChild("text")->getContent());
+		} else if (strctNode->getName() == "strref") {
+			uint32 value;
+			Common::parseString(strctNode->getProperty("strref"), value);
+			strctPtr->addStrRef(strctNode->getProperty("label"), value);
+		} else if (strctNode->getName() == "resref") {
+			if (strctNode->findChild("text") == 0)
+				strctPtr->addResRef(strctNode->getProperty("label"), "");
+			else
+				strctPtr->addResRef(strctNode->getProperty("label"), strctNode->findChild("text")->getContent());
+		} else if (strctNode->getName() == "void") {
+			strctPtr->addVoid(
+				strctNode->getProperty("label"),
+				reinterpret_cast<const byte*>(strctNode->findChild("text")->getContent().c_str()),
+				strctNode->findChild("text")->getContent().size()
+			);
+		} else if (strctNode->getName() == "vector") {
+			float x, y, z;
+
+			if (strctNode->getChildren().size() != 3)
+				throw Common::Exception("GFF3Creator::readStructContents() Invalid size of vector components");
+
+			XMLNode::Children::const_iterator iter = strctNode->getChildren().begin();
+			Common::UString xValue = (*iter)->getContent();
+			++iter;
+			Common::UString yValue = (*iter)->getContent();
+			++iter;
+			Common::UString zValue = (*iter)->getContent();
+
+			Common::parseString(xValue, x);
+			Common::parseString(yValue, y);
+			Common::parseString(zValue, z);
+
+			strctPtr->addVector(strctNode->getProperty("label"), x, y, z);
+		} else if (strctNode->getName() == "orientation") {
+			float x, y, z, w;
+
+			if (strctNode->getChildren().size() != 4)
+				throw Common::Exception("GFF3Creator::readStructContents() Invalid size of orientation components");
+
+			XMLNode::Children::const_iterator iter = strctNode->getChildren().begin();
+			Common::UString xValue = (*iter)->getContent();
+			++iter;
+			Common::UString yValue = (*iter)->getContent();
+			++iter;
+			Common::UString zValue = (*iter)->getContent();
+			++iter;
+			Common::UString wValue = (*iter)->getContent();
+
+			Common::parseString(xValue, x);
+			Common::parseString(yValue, y);
+			Common::parseString(zValue, z);
+			Common::parseString(wValue, w);
+
+			strctPtr->addOrientation(strctNode->getProperty("label"), x, y, z, w);
+		} else if (strctNode->getName() == "locstring") {
+			uint32 strref;
+			Aurora::LocString locString;
+
+			Common::parseString(strctNode->getProperty("strref"), strref);
+			locString.setID(strref);
+
+			if (!strctNode->getChildren().empty()) {
+				for (const auto &child : strctNode->getChildren()) {
+					if (child->getName() != "string")
+						throw Common::Exception("GFF3Creator::readStructContents() Invalid LocString string");
+
+					uint32 language;
+					Common::parseString(child->getProperty("language"), language);
+
+					// TODO: Add gender if available.
```

Comment:
Can you fulfill this TODO first? I'd rather not have a tool available that accidentally removes information when doing a round-trip